src/ncp/ncp_host.cpp

```diff
@@ -80,10 +80,12 @@ const char *NcpHost::GetCoprocessorVersion(void)
 
 void NcpHost::Init(void)
 {
+    otbr::NetifIp6Implementor implementor = {
+        [this](const uint8_t *aData, uint16_t aLength) { return mNcpSpinel.Ip6Send(aData, aLength); }};
```

Comment:
In AOSP, there's a pattern of making its dependencies a class which can help reduce the number of parameters and make the class more testable. I wonder if we can borrow the idea here.

https://cs.android.com/android/platform/superproject/main/+/main:packages/modules/Connectivity/service/src/com/android/server/connectivity/MulticastRoutingCoordinatorService.java;l=761;drc=5005c919c7cd6d27edad9954785d8f59ae6f78fb

Reply:
Thanks! I think this is a good model. I have updated to use virtual interface as the dependencies class.